mitiq/lre/multivariate_scaling/layerwise_folding.py

```diff
@@ -0,0 +1,195 @@
+
+    if num_chunks > num_layers:
+        raise ValueError(
+            "Number of chunks > the number of layers in the circuit."
+        )
+
+    k, m = divmod(num_layers, num_chunks)
+    split_circuit = (
+        input_circuit[i * k + min(i, m) : (i + 1) * k + min(i + 1, m)]
+        for i in range(num_chunks)
+    )
+    return list(split_circuit)
+
+
+def _get_scale_factor_vectors(
+    input_circuit: Circuit,
+    degree: int,
+    fold_multiplier: int,
+    num_chunks: Optional[int] = None,
+) -> List[Tuple[Any, ...]]:
+    """Returns the patterned scale factor vectors required for multivariate
+    extrapolation.
+
+        Args:
+            input_circuit: Circuit to be scaled
+            degree: Degree of the multivariate polynomial
+            fold_multiplier: Scaling gap required by unitary folding
+
+        Returns:
+            scale_factor_vectors: Multiple variations of scale factors for each
+                layer in the input circuit
+    """
+
+    circuit_chunks = _get_chunks(input_circuit, num_chunks)
+    num_layers = len(circuit_chunks)
+
+    # find the exponents of all the monomial terms required for the folding
+    # pattern
+    pattern_full = []
+    for i in range(degree + 1):
+        for j in itertools.combinations_with_replacement(range(num_layers), i):
+            pattern = [0] * num_layers
+            # get the monomial terms in graded lexicographic order
+            for index in j:
+                pattern[index] += 1
+            # use fold multiplier on the folding pattern to figure out which
+            # layers will be scaled
+            pattern_full.append(tuple(fold_multiplier * np.array(pattern)))
+
+    # get the scale factor vectors
+    # the layers are scaled as 2n+1 for unitary folding
+    return [
+        tuple(2 * num_folds + 1 for num_folds in pattern)
+        for pattern in pattern_full
+    ]
+
+
+def multivariate_layer_scaling(
+    input_circuit: Circuit,
+    degree: int,
+    fold_multiplier: int,
+    num_chunks: Optional[int] = None,
+    folding_method: Callable[
+        [QPROGRAM, float], QPROGRAM
+    ] = fold_gates_at_random,
+) -> List[Circuit]:
+    r"""
+    Defines the noise scaling function required for Layerwise Richardson
+    Extrapolation as defined in :cite:`Russo_2024_LRE`.
+
+    Args:
+        input_circuit: Circuit to be scaled
+        degree: Degree of the multivariate polynomial
+        fold_multiplier: Scaling gap required by unitary folding
+        num_chunks: Number of desired approximately equal chunks. When the
+            number of chunks is the same as the layers in the input circuit,
+            the input circuit is unchanged.
+        folding_method: Unitary folding method. Default is
+            :func:`fold_gates_at_random`
+
+    Returns:
+        Multiple folded variations of the input circuit
+
+    Raises:
+        ValueError:
+            When the degree for the multinomial is not >= to 1.
```

Comment:
It's a bit confusing to read the `ValueError` comment and see the logic in the code as they are worded in a way that is stated as the negation of the thing being checked. Perhaps word these in the same way in which the logic is being invoked to check the condition?

So, for instance, this would be

        ValueError:
            When the degree for the multinomial is < 1

etc.